torch_xla/torch_xla.py

```diff
@@ -59,20 +59,14 @@ def device_count() -> int:
   return len(real_devices())
 
 
-def sync(wait: bool = False):
+def sync(wait: bool = False, reset_scope: bool = True):
   """Launches all pending graph operations.
 
   Args:
     wait (bool): whether to block the current process until the execution finished.
-
+    reset_scope (bool): whether to reset the torch::lazy::ScopeContext of the IR Nodes.
   """
-  torch_xla._XLAC._xla_step_marker(
-      torch_xla._XLAC._xla_get_default_device(),
-      [],
-      wait=wait,
-  )
-  devctx = xm._run_step_closures()
-  torch_xla._XLAC._set_all_reduce_token(devctx.device, None)
+  xm.mark_step(wait, reset_scope)
```

Comment:
The goal is to deprecate and remove mark_step. If you call mark_step here then you cannot delete mark_step.

Recommendation:

1. inline the body of mark_step here.
2. mark_step prints deprecated message and calls sync
3. replace all references of mark_step to sync in our tests / docs
4. (in next release) delete mark_step.

Reply:
you can probably do 1 + 2 + 3 or 1 + 2 in this PR, (do 3 in a separate PR if latter). 4 need to happen in a different release because its BC breaking.

Reply:
Makes sense for 1 and 2. I have #9086 doing 3, but I may need to do 3 in multiple PRs to make debugging easier. 4 sounds good as well.